lib/govuk_publishing_components/presenters/breadcrumbs.rb

```diff
@@ -20,15 +20,68 @@ def structured_data
 
       def item_list_element
         breadcrumbs.each_with_index.map do |crumb, index|
-          {
-            "@type" => "ListItem",
-            "position" => index + 1,
-            "item" => list_item_item(crumb)
+          Breadcrumb.new(crumb, index).item_list_element(@request_path)
+        end
+      end
+    end
+
+    class Breadcrumb
+      extend Forwardable
+      def_delegators :@crumb, :[]
+
+      def initialize(crumb, index)
+        @crumb = crumb
+        @index = index + 1
+      end
+
+      def item_list_element(request_path)
+        {
+          "@type" => "ListItem",
+          "position" => index,
+          "item" => list_item_item(request_path)
+        }
+      end
+
+      def is_link?
+        crumb[:url].present? || crumb[:is_current_page]
+      end
+
+      def path
+        crumb[:is_current_page] ? '#content' : crumb[:url]
+      end
+
+      def aria_current
+        crumb[:is_current_page] ? 'page' : 'false'
+      end
+
+      def tracking_data(breadcrumbs_length)
+        data = {
+          track_category: 'breadcrumbClicked',
+          track_action: index,
+          track_label: path,
+          track_options: {
+              dimension28: breadcrumbs_length.to_s,
+              dimension29: crumb[:title]
           }
+        }
+
+        is_homepage = crumb[:url] == "/"
+
+        if is_homepage
+          data[:track_category] = 'homeLinkClicked'
+          data[:track_action] = 'homeBreadcrumb'
+          data[:track_label] = ''
```

Comment:
It's perhaps worth having this for clarity - it'll add an empty attribute of `data-track-label` to the element but this shouldn't break anything.

Reply:
Haur wanted it to be a blank string so it does need to be changed...